azkaban-exec-server/src/main/java/azkaban/container/ContainerServlet.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.container;
+
+import azkaban.Constants;
+import azkaban.executor.ConnectorParams;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.server.HttpRequestUtils;
+import azkaban.utils.FileIOUtils;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import static azkaban.common.ServletUtils.writeJSON;
+import static azkaban.server.HttpRequestUtils.hasParam;
+import static azkaban.server.HttpRequestUtils.getParam;
+import static azkaban.server.HttpRequestUtils.getIntParam;
+
+
+public class ContainerServlet extends HttpServlet implements ConnectorParams {
```

Comment:
Can you please validate whether ContainerServlet can extend LoginAbstractAzkabanServlet? Authentication will come automatically. With that, LoginAbstractAzkabanServlet also extends HttpServlet indirectly so it should work well.

Reply:
Tried to do this, but there are a bunch of dependency conflicts that show up. Let's create a separate ticket for this. After merge with Deepak's code, this can be taken up.

Reply:
Sounds good. Thank you for looking into it.